### PR TITLE
feat: `ignorePath` option

### PR DESCRIPTION
**Pull request by evilebottnawi**
**Description of change**

Fixes https://github.com/tclindner/npm-package-json-lint/issues/99

**Checklist**

  - [x] Unit tests have been added
  - [x] Specific notes for documentation, if applicable

### CODE REVIEW

File: src/CLIEngine.js [hunk 1]
```diff
@@ -197,12 +226,13 @@ const getFileList = function(patterns, options) {
...
   const files = [];
   const addedFiles = new Set();
+  const ignorer = getIgnorer(cwd, options);
 
   globPatterns.forEach((pattern) => {
     const file = path.resolve(cwd, pattern);
 
     if (fs.existsSync(file) && fs.statSync(file).isFile()) {
-      if (addedFiles.has(file)) {
+      if (addedFiles.has(file) || ignorer.ignores(pattern)) {
```
Comment (tclindner) [inline src/CLIEngine.js]: It looks like `ignorer.ignores()` takes a path name. I wonder if it is having trouble accepting a glob pattern. What do you think about passing in `file` instead. `file` is getting resolved based on the `cwd` and the `pattern` on line 232.

**Comment by tclindner** [comment]
Hey @evilebottnawi! Thank you so much for opening this PR 🙌 I don't see the `TypeError` you referenced above. It looks like the tests are failing with the following error. I added a comment in the PR about tweaking the path passed to the ignorer. Let me know what you think!
`RangeError: path should be`

I took a peek at master and it looks like it is green. 
image

**Comment by evilebottnawi** [comment]
@tclindner hm, i can't reproducible this problem locally, one problem for me is `ypeError: Cannot read property 'equal' of undefined` :confused: And same problem in `master` (reinstall deps doesn't solve problem), investigate

**Comment by evilebottnawi** [comment]
It is really weird - `npm test` and `npm coverage` have difference behavior :confused:

**Comment by evilebottnawi** [comment]
Yep, in some cases we have absolute path, let's use `path.relative` to avoid this problem. All green :+1:

**Pull request merged**